app/controllers/courses_controller.rb

```diff
@@ -70,6 +70,7 @@ def show
     protect_privacy
     verify_edit_credentials { return }
     set_enrollment_details_in_session
+    @tracking_description = TrackingDescriptionBuilder.new(@course).call
```

Comment:
I think this would be better as an instance method available to the Course object (which can then be called directly from the jbuilder template without any changes to the controller). 

The service still makes sense as a separate file, but it can be called from the Course method. We do similar things with TrainedStudentsManager and other classes that are responsible for generating some kind of data starting from a course.